controllers/certificate/certificate.go

```diff
@@ -0,0 +1,185 @@
+package certificate
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	netutils "k8s.io/utils/net"
+)
+
+func generateSelfSignedCertificate(host string, alternateIPs []net.IP, alternateDNS []string, maxAge time.Duration) ([]byte, []byte, error) {
+	validFrom := time.Now().Add(-time.Hour) // valid an hour earlier to avoid flakes due to clock skew
```

Comment:
```suggestion
	validFrom := time.Now().Sub(time.Hour) // valid an hour earlier to avoid flakes due to clock skew
```

Reply:
I thought the same, but I think someone just messed up the API 😬 
Sub has Time as param([docs](https://pkg.go.dev/time#Time.Sub)). In the docs there is even a proposal to use Add(-Duration):
> To compute t-d for a duration d, use t.Add(-d).

Reply:
Wow